Replace deprecated request lib with got

lutangar opened the pull request:
This simplifies the a lot the `browser` and bring retry behavior by default.
Fix #6
But also simplify a lot of the API code overall *less code, less bugs*

Testing `lastRequest` - at least the way it was - was completely flawed and thus was removed regarding the asynchronous nature of Js. Fix #16

This add *kind of* behavior tests via `test/sandbox.spec.js` runnable via for example via `API_KEY=xxx yarn test:sandbox` further looking for possible regressions.

This is almost done, just need to clean the way errors are logged and look into the release process but I would appreciate a review @olivierapivideo @dougsillars at least to see if I forgot something behind.

I also deprecated methods which are non existent endpoints (I wanted to test these but realized once again taht they simply shouldn't exists) but I might as well do that in another PR.

olivierapivideo reviewed (approved):
it looks good to me :+1:
great simplification (but I think I will miss all the "thats" ;) )

lutangar commented:
I added an `errorHandler` and make "large file" upload works with the new library.

Few notices on file upload though:
- I removed the `lastchunkextension` header check simply because the API doesn't send it anymore
- Error sent by the API in case of failed download point to a broken URL:
```
{
  type: 'https://docs.api.video/docs/zerofileuploaded',
  title: 'If the file size is larger than 128Mb, please follow the documentation for big files.',
  name: 'file',
  'recommended-documentation': 'https://docs.api.video/5.1/vod-tutorial/step-3-upload-a-video-file#example-2-big-file-multiple-requests',
  status: 400
}
```
